frontend/fingertips-frontend/lib/search/indicatorSearchService.ts

```diff
@@ -24,16 +24,28 @@ export class IndicatorSearchService implements IIndicatorSearchService {
     );
   }
 
-  async searchWith(searchTerm: string): Promise<IndicatorDocument[]> {
+  async searchWith(
+    searchTerm: string,
+    areaCodes?: string[]
+  ): Promise<IndicatorDocument[]> {
     // Search with both full search term and wildcard for now to allow for
     // observed behaviour where wildcard match of full term did not always return
     // the matching document.
     const query = `${searchTerm} /.*${searchTerm}.*/`;
 
+    // This creates an AI Search filter string which should look like
+    //  associatedAreaCodes/any(a: a eq 'E09000023' or a eq 'E09000013' or a eq 'E09000025')
+    const formatFilterString = (areaCodes: string[]) => {
+      if (areaCodes.length == 0) return undefined;
+      const areaCodeEqualityStrings = areaCodes.map((a) => `a eq '${a}'`);
+      return `associatedAreaCodes/any(a: ${areaCodeEqualityStrings.join(' or ')})`;
+    };
+
     const searchOptions: SearchOptions<IndicatorDocument> = {
       queryType: 'full',
       includeTotalCount: true,
       top: 100,
+      filter: areaCodes ? formatFilterString(areaCodes) : undefined,
```

Comment:
Do we need to update the tests in `indicatatorSearchService` to check the `searchOptions` are passed as expected?